Api review

xavluiz opened the pull request:
Please review the api function names. Some of these are legacy and may need to remain due to current user usage.

xavluiz committed:
branch update

xavluiz committed:
branch update

o4sw commented on index.ts:
```diff
@@ -21,34 +22,34 @@ export function setCredentials(credentials: any) {
+ * @param player {spotify|spotify-web|itunes }
+ * @returns {Promise<boolean> }
  */
 export async function isRunning(player: PlayerName) {
```
"isPlayerRunning" (less ambiguous).

Also, in the description, consider "Checks if a player is running"

o4sw commented on index.ts:
```diff
@@ -58,40 +59,40 @@ export async function isRunning(player: PlayerName) {
+/** review name
  * Kills the Spotify desktop if it's running
  */
 export function stopSpotifyIfRunning() {
```
I think "stopSpotify" is enough ... (whether it was running or not could be part of a return value???)

o4sw commented on index.ts:
```diff
@@ -58,40 +59,40 @@ export async function isRunning(player: PlayerName) {
+/** review name
  * Kills the iTunes desktop if it's running
  */
 export function stopItunesIfRunning() {
```
I think "stopItunes" is enough ... (whether it was running or not could be part of a return value???)

o4sw commented on index.ts:
```diff
@@ -58,40 +59,40 @@ export async function isRunning(player: PlayerName) {
+/** review name
  * Launches the Spotify desktop if it's not running
  */
 export function startSpotifyIfNotRunning() {
```
I think "startSpotify" is enough (whether it was running or not could be part of the return value???)

o4sw commented on index.ts:
```diff
@@ -58,40 +59,40 @@ export async function isRunning(player: PlayerName) {
+/** review name
  * Launches the iTunes desktop if it's not running
  */
 export function startItunesIfNotRunning() {
```
I think "startItunes" is enough (whether it was running or not could be part of the return value???)

o4sw commented on index.ts:
```diff
@@ -58,40 +59,40 @@ export async function isRunning(player: PlayerName) {
- * @returns {artist, album, genre, disc_number, duration, played_count, track_number, id, name, state}
+ * @returns {artist, album, genre, disc_number, duration, played_count, track_number, id, name, state }
  */
 export async function getState(player: PlayerName): Promise<TrackState> {
```
"getPlayerState" might be better, especially if there is some chance that there would be a need for a "getState" function which has some other meaning.

o4sw commented on index.ts:
```diff
@@ -154,18 +155,18 @@ export function playTrackInContext(player: PlayerName, params: any[]) {
+ * @param device_id {string }
+ * @param play {boolean }
  */
 export function playSpotifyDevice(device_id: string, play: boolean = true) {
```
This is probably ok, but confuses me a little b/c the name and description seem to not exactly match the intent.
If you pass `false` for the play param what would happen? In other words, if the function is described as initiate and play, then maybe it should not even take the second param, just play.

o4sw commented on index.ts:
```diff
@@ -233,42 +234,42 @@ export function previous(player: PlayerName, options: any) {
 export function setShufflingOff(player: PlayerName) {
     return musicCtr.run(player, "setShuffling", ["false"]);
 }
 
```
maybe only expose one api: `setShuffle(player: PlayerName, shuffle: boolean)` ???

o4sw commented on index.ts:
```diff
@@ -291,59 +292,59 @@ export async function isRepeating(player: PlayerName) {
- * @param player {spotify|spotify-web|itunes}
+ * @param player {spotify|spotify-web|itunes }
  */
 export async function playlistNames(player: PlayerName) {
```
`getPlaylistNames`

o4sw reviewed (approved):
These are only a superficial review. I may not be aware of other reasons for some of the names.

Jedi96 reviewed (approved):
looks good, thanks.

geoffstevens8 commented on index.ts:
```diff
@@ -233,42 +234,42 @@ export function previous(player: PlayerName, options: any) {
  */
 export function setShufflingOn(player: PlayerName) {
     return musicCtr.run(player, "setShuffling", ["true"]);
 }
```
You might consider making the shuffling functions similar to repeat functions (e.g. pick either repeatOn and shufflingOn OR setRepeatOn and setShufflingOn)

geoffstevens8 commented on index.ts:
```diff
@@ -291,59 +292,59 @@ export async function isRepeating(player: PlayerName) {
  */
 export function unMute(player: PlayerName) {
     return musicCtr.run(player, "unMute");
 }
```
All lowercase "unmute" might feel more natural, as its one word

xavluiz commented:
> I would say running === app, playing === song

where in the code is this? the other comments show the code above the comments. i'm not sure what you're referring to

xavluiz commented:
> Is this for itunes?
> 
> It might be tricky to use as a dev, as sometimes there is one method for a function (mute/unmute) and sometimes it's player specific. Just something to think about.

again on this one, which code?

xavluiz commented:
> Comment makes it seems like it would kill the actual process rather than stopping playback. Function name makes sense though.

I think i know what you're referring to now. I'm moving those into the util. those (kill and launch) were only meant for the tests. thank you for pointing them out

xavluiz committed:
updated method names and comments based on feedback

desktop player tests
    ✓ Should Show The Playlist Names AND Show The Tracks Of A Playlist (2006ms)
    ✓ Should Set An Itunes Song's Love State (4130ms)
    ✓ Should Show An Error (1179ms)
    ✓ Should Show Spotify Is Not Running (83ms)
    ✓ Should Show Itunes Is Not Running (79ms)
    ✓ Get Spotify Track Info (5151ms)
    ✓ Get iTunes Track Info (3784ms)
failed to run itunes tests:  Invalid Chai property: equal.to. See docs for proper usage of "equal".
    ✓ Tell Spotify To Perform Next, Repeat, Previous, Pause, Shuffle, Volume Change, Mute, Unmute, Shuffle, Check Shuffle (7269ms)
failed to run itunes tests:  CodyMusic.repeatOn is not a function
    ✓ Tell Itunes To Perform Next, Repeat, Previous, Pause, Shuffle, Volume Change, Mute, Unmute, Shuffle, Check Shuffle (5157ms)

  feature api tests
    ✓ Fetch genre from itunes using artist and song (278ms)
    ✓ Fetch genre from itunes using artist (669ms)
    ✓ Fetch genre from spotify using artist (595ms)

  web player music tests
    ✓ Check the spotify web player devices (105ms)
    ✓ Play on spotify device (3289ms)
    ✓ Pause web player (3244ms)
    ✓ Play specific track and validate track is playing (3396ms)
    ✓ Go to the next and previous track (5721ms)

  17 passing (48s)

xavluiz commented:
I've updated the api names and moved the kill and launch app methods to the controller. Thanks for the feedback. Let me know if there's anything else you see. I'm happy to update it. Updated the test cases as well.

desktop player tests
    ✓ Should Show The Playlist Names AND Show The Tracks Of A Playlist (2006ms)
    ✓ Should Set An Itunes Song's Love State (4130ms)
    ✓ Should Show An Error (1179ms)
    ✓ Should Show Spotify Is Not Running (83ms)
    ✓ Should Show Itunes Is Not Running (79ms)
    ✓ Get Spotify Track Info (5151ms)
    ✓ Get iTunes Track Info (3784ms)
failed to run itunes tests:  Invalid Chai property: equal.to. See docs for proper usage of "equal".
    ✓ Tell Spotify To Perform Next, Repeat, Previous, Pause, Shuffle, Volume Change, Mute, Unmute, Shuffle, Check Shuffle (7269ms)
failed to run itunes tests:  CodyMusic.repeatOn is not a function
    ✓ Tell Itunes To Perform Next, Repeat, Previous, Pause, Shuffle, Volume Change, Mute, Unmute, Shuffle, Check Shuffle (5157ms)

  feature api tests
    ✓ Fetch genre from itunes using artist and song (278ms)
    ✓ Fetch genre from itunes using artist (669ms)
    ✓ Fetch genre from spotify using artist (595ms)

  web player music tests
    ✓ Check the spotify web player devices (105ms)
    ✓ Play on spotify device (3289ms)
    ✓ Pause web player (3244ms)
    ✓ Play specific track and validate track is playing (3396ms)
    ✓ Go to the next and previous track (5721ms)


  17 passing (48s)

xavluiz committed:
added the features api
